sys/net/link_layer/iQueue-MAC/simple.c

```diff
@@ -0,0 +1,2082 @@
+typedef enum {     
+    IQUEUE_BEACON_MANAGE,  
+    IQUEUE_VTDMA_MANAGE,   
+    IQUEUE_SP_MANAGE,   
+    IQUEUE_CP_MANAGE,   
+    IQUEUE_PREAMBLE_MANAGE,
+    IQUEUE_TP_MANAGE,   
+    
+    IQUEUE_SIMPLE_GOT_BEACON,
+    IQUEUE_SIMPLE_TDMA_MANAGE,
+    IQUEUE_SIMPLE_SP_MANAGE,
+    IQUEUE_SIMPLE_CP_MANAGE,
+    IQUEUE_GET_NEW_PACKET,
+    IQUEUE_EMIT_PACKET,
+    IQUEUE_WAIT,    
+    //IQUEUE_SIMPLE_CHECK_DATA,
+    IQUEUE_STOP
+} iqueue_msg_t;
+
+
+/* queue of packets to be send during next transmission period */
+typedef struct out_pkt {
+    struct out_pkt *next;    
+    void *pkt_data;
+    unsigned int data_len;
+#if PERFORM_EXTENDED_CALLBACKS
+    mac_tx_status_callback_t callback;
+    void *callback_param;
+#endif /* PERFORM_EXTENDED_CALLBACKS */
+} out_pkt_t;
+static out_pkt_t outgoing_queue[MAX_OUTGOING_PACKETS_NUMBER];
+static out_pkt_t * volatile first_outgoing_packet;
+
+typedef struct out_pkt_buffer {    
+   u8 pkt_data_buf[MAX_DATA_LENGTH]    
+} out_pkt_t_buffer;
+
+static out_pkt_t_buffer outgoing_queue_buf[MAX_OUTGOING_PACKETS_NUMBER];
+static u8 queue_buffer_tail=0;
+
+
+/* Callback function to relay incoming packets to other net layers or apps */
+static mac_rx_packet_notif_t rx_callback_func = NULL;
+
+
+/* flag indicating whether we can transmit in burst (fast) mode,
+   or whether we must use classical CSMA/CA method. */
+static bool in_burst_mode = false;
+
+/* number of backoffs waited for accessing channel; used in CSMA/CA mode */
+static int num_backoffs = 0;
+
+/* number of attempted transmission for the packet currently emitted;
+   used in CSMA/CA mode */
+static int num_tx = 0;
+
+
+/* iQueue-MAC parameters */
+u8 iqueue_mac_mode = SIMPLE_MODE; ///  ROUTER_MODE; //                                
+
+u8 *rx_data_buf;
+
+//////Beacon manage parameters
+extern u8 beacon_buf[120];
+//////Subframe manage parameters
+extern u32 subframeduration;
+extern u32 cp_duration_us;
+extern u32 tdma_duration;
+extern u32 sp_duration;
+extern bool vtdma_exist;
+extern node_regis_struc node_regis_list[MAX_LOCAL_NODE_NUM];
+
+hwtimer_ticks_t sp_start_time_t = 0;
+//////SP manage parameters
+bool sp_turn_to_sleep_state = false;
+bool sp_end = false;
+//////CP manage parameters
+volatile bool cp_got_new_packet = false;
+//////TP manage parameters
+u8 tp_preamble_counter = MAX_PREAMBLE_MUN;
+bool tp_get_preamble_ack = false;
+bool tp_start_trans_preamble = false;
+bool channel_busy = false;
+u8 preamble_buf[50];
+
+unsigned short neighbor_router_ID[MAX_NEIGHBOR_ROUTER_NUM]; 
+
+
+static volatile u8 mac_state = 0;
+
+ieee802154_node_addr_t iq_dest;
+ieee802154_node_addr_t router_dest;
+
+
+/* iQueue-MAC parameters - simple node*/
+volatile u8 mac_queue_length = 0;
+u32 simple_slots_start_time_us = 0;
+static bool simple_node_get_beacon = false;
+static bool simple_node_get_slots = false;
+
+u16 mac_short_address = 0;
+u8 simple_slots_num = 0;
+volatile bool end_csma_backoff = false;
+
+
+double Drand(void);
```

Comment:
>> +double Drand(void);

Also, if possible, avoid double, as it will, even if supported, pull in
a lot of extra code.